internal/mode/static/handler_test.go

```diff
@@ -132,6 +147,72 @@ var _ = Describe("eventHandler", func() {
 		})
 	})
 
+	When("receiving control plane configuration updates", func() {
+		cfg := func(level nkgAPI.ControllerLogLevel) *nkgAPI.NginxGateway {
+			return &nkgAPI.NginxGateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      configName,
+				},
+				Spec: nkgAPI.NginxGatewaySpec{
+					Logging: &nkgAPI.Logging{
+						Level: helpers.GetPointer(level),
+					},
+				},
+			}
+		}
+
+		expStatuses := func(cond conditions.Condition) status.Statuses {
+			return status.Statuses{
+				NginxGatewayStatus: &status.NginxGatewayStatus{
+					NsName: types.NamespacedName{
+						Namespace: namespace,
+						Name:      configName,
+					},
+					Conditions:         []conditions.Condition{cond},
+					ObservedGeneration: 0,
+				},
+			}
+		}
+
+		It("handles a valid config", func() {
+			batch := []interface{}{&events.UpsertEvent{Resource: cfg(nkgAPI.ControllerLogLevelError)}}
+			handler.HandleEventBatch(context.Background(), batch)
+
+			Expect(fakeStatusUpdater.UpdateCallCount()).Should(Equal(1))
+			_, statuses := fakeStatusUpdater.UpdateArgsForCall(0)
+			Expect(statuses).To(Equal(expStatuses(staticConds.NewNginxGatewayValid())))
+			Expect(handler.cfg.logLevelSetter.Enabled(zap.DebugLevel)).To(BeFalse())
```

Comment:
would it make sense to check it for other cases too?

Reply:
Enabled works by returning if the level is at or above the current level, so checking other levels is redundant.

Reply:
oh, sorry. I mean for other cases  - handles an invalid config, handles a deleted config"

Reply:
Those tests are below this one.

Reply:
yep. so because those cases involve setting the log level to the default, can we test that?

so inside `It("handles an invalid config", func()` { and `It("handles a deleted config", func() {`

Reply:
Oh I see